### PR TITLE
Fixes #9888 - use random_password over generate_password

### PR DESCRIPTION
**Pull request by dustints**
generate_password has a dependency on openssl.

### CODE REVIEW

File: manifests/init.pp [hunk 1]
```diff
@@ -120,7 +120,7 @@
   $ca_key = "${certs::pki_dir}/private/${default_ca_name}.key"
   $ca_cert = "${certs::pki_dir}/certs/${default_ca_name}.crt"
   $ca_cert_stripped = "${certs::pki_dir}/certs/${default_ca_name}-stripped.crt"
-  $ca_key_password = cache_data('ca_key_password', generate_password())
+  $ca_key_password = cache_data('ca_key_password', random_password())
```
Comment (stbenjam) [inline manifests/init.pp]: random_password takes a length argument, and you should bump the foreman requirement in metadata.json to >= 2.0.0, as that is when it was introduced

Comment (dustints) [inline manifests/init.pp]: updated

File: manifests/init.pp [hunk 2]
```diff
@@ -120,7 +120,7 @@
   $ca_key = "${certs::pki_dir}/private/${default_ca_name}.key"
   $ca_cert = "${certs::pki_dir}/certs/${default_ca_name}.crt"
   $ca_cert_stripped = "${certs::pki_dir}/certs/${default_ca_name}-stripped.crt"
-  $ca_key_password = cache_data('ca_key_password', generate_password())
+  $ca_key_password = cache_data('ca_key_password', random_password(8))
```
Comment (stbenjam) [inline manifests/init.pp]: Nitpicky, but 8 is very short.  I'd say 24 at least.  No one's going to be typing this in

**Comment by dustints** [comment]
@stbenjam sounds good, updated!

**Comment by stbenjam** [comment]
ACK

**Pull request merged**